src/google/protobuf/compiler/parser_unittest.cc

```diff
@@ -1977,8 +1977,88 @@ TEST_F(ParserValidationErrorTest, Proto3JsonConflictError) {
       "  uint32 foo = 1;\n"
       "  uint32 Foo = 2;\n"
       "}\n",
-      "3:9: The JSON camel-case name of field \"Foo\" conflicts with field "
-      "\"foo\". This is not allowed in proto3.\n");
+      "3:9: The default JSON name of field \"Foo\" (\"Foo\") conflicts "
+      "with the default JSON name of field \"foo\" (\"foo\"). "
+      "This is not allowed in proto3.\n");
+}
+
+TEST_F(ParserValidationErrorTest, Proto2JsonConflictError) {
+  // conflicts with default JSON names are not errors in proto2
+  ExpectParsesTo(
```

Comment:
Can you use C++ raw strings here?

Reply:
I can. But you'll notice that all of the test cases above and below use this format of string constants. I was going for consistency.

Reply:
Holding off on this, just because it would make these new tests look very different from all the rest. Let me know if you insist on this change or not.

Reply:
I would prefer that we update this test case, even if it makes it look different from existing ones. I have been gradually burning down old-style strings throughout the codebase.